roles/postgres_replicant/tasks/main.yml

```diff
@@ -18,9 +18,19 @@
     - database
     - replication
 
+- name: check that replication connection is possible
+  become: yes
+  become_user: postgres
+  command: psql 'dbname=postgres replication=database host={{ hostvars[groups.database[0]].ansible_default_ipv4.address }} user={{ replicator_db_user|default(default_replicator_db_user) }}' -c 'IDENTIFY_SYSTEM;'
+  register: replication_connection
+  tags:
+    - database
+    - replication
+
 - when:
     - "'f' in replication_running.stdout"
     - not recoveryconf_file.stat.exists
+    - "'systemid' in replication_connection.stdout"
```

Comment:
Hmm... is it possible that the main service hasn't fully restarted yet and that is why it's not ready for the replication connection?

You can deliberately stop the play using the [fail module](https://docs.ansible.com/ansible/fail_module.html), but we don't really want to do this because it can interrupt handlers that may have already been triggered. Though, that is not the case for this role at this time.